### ISSUE TITLE
Tutorial error: Uncaught TypeError: this.lowLevel(...).root is not a function

### ISSUE THREAD

**Issue by drernie** [mulesoft/api-workbench#325]
[Enter steps to reproduce below:]

While going through the Getting Started Tutorial
1. Put cursor at Mammal type and activate Pull up action from Palette or context menu
2. Move color property from left to right, select Pet supertype target and press Extract button.

![screen shot 2016-10-06 at 1 44 34 pm](https://cloud.githubusercontent.com/assets/19791/19169897/1f450322-8bcb-11e6-8eeb-85dd63d4b64b.png)

**Atom Version**: 1.10.2
**Electron Version**: 0.37.8
**System**: Mac OS X 10.12
**Thrown From**: [api-workbench](https://github.com/mulesoft/api-workbench) package, v0.8.38

### Stack Trace

Uncaught TypeError: this.lowLevel(...).root is not a function

```
At /Users/ernest/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/highLevelImpl.js:1531

TypeError: this.lowLevel(...).root is not a function
    at ASTNodeImpl.isEmptyRamlFile (/Users/ernest/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/highLevelImpl.js:1531:38)
    at Object.addToNode (/Users/ernest/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/ast.core/mutators.js:347:36)
    at ASTNodeImpl.add (/Users/ernest/.atom/packages/api-workbench/node_modules/raml-1-parser/dist/raml1/highLevelImpl.js:1381:18)
    at e.t.mergeNode (/Users/ernest/.atom/packages/api-workbench/main.js:3:28639)
    at /Users/ernest/.atom/packages/api-workbench/main.js:3:28237
    at Array.forEach (native)
    at e.t.mergeTrees (/Users/ernest/.atom/packages/api-workbench/main.js:3:28215)
    at e.performOk (/Users/ernest/.atom/packages/api-workbench/main.js:4:8488)
    at /Users/ernest/.atom/packages/api-workbench/main.js:3:27778
    at /Users/ernest/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:413:68
    at Array.forEach (native)
    at HTMLButtonElement.element.onclick (/Users/ernest/.atom/packages/api-workbench/node_modules/atom-ui-lib/dist/UI.js:413:31)
```

### Commands

```
     -1:28.7.0 editor:indent-selected-rows (atom-text-editor.editor.is-focused)
     -1:21.4.0 editor:newline (atom-text-editor.editor.is-focused)
  2x -1:19.5.0 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
  3x -1:05.4.0 core:backspace (atom-text-editor.editor.is-focused)
  2x -1:04.2.0 editor:newline (atom-text-editor.editor.is-focused)
  5x -1:00.8.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:59.4.0 editor:newline (atom-text-editor.editor.is-focused)
     -0:51.6.0 core:copy (atom-text-editor.editor.is-focused)
     -0:51.2.0 core:move-down (atom-text-editor.editor.is-focused)
  2x -0:50.7.0 core:move-up (atom-text-editor.editor.is-focused)
     -0:49.5.0 core:paste (atom-text-editor.editor.is-focused)
  4x -0:46 editor:outdent-selected-rows (atom-text-editor.editor.is-focused)
  4x -0:41.4.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:32.6.0 core:paste (atom-text-editor.editor.is-focused)
     -0:29.7.0 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
  6x -0:27.5.0 core:backspace (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "raml"
    ],
    "projectHome": "/Users/ernest/Developer/",
    "themes": [
      "one-dark-ui",
      "base16-tomorrow-dark-theme"
    ]
  }
}
```

### Installed Packages

```coffee
# User
api-workbench, v0.8.38 (active)
git-time-machine, v1.5.3 (inactive)
linter, v1.11.18 (active)
raml, v0.2.0 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.44.0 (inactive)
base16-tomorrow-dark-theme, v1.1.0 (active)
base16-tomorrow-light-theme, v1.1.1 (inactive)
one-dark-ui, v1.5.0 (active)
one-light-ui, v1.5.0 (inactive)
one-dark-syntax, v1.3.0 (inactive)
one-light-syntax, v1.3.0 (inactive)
solarized-dark-syntax, v1.0.2 (inactive)
solarized-light-syntax, v1.0.2 (inactive)
about, v1.5.3 (active)
archive-view, v0.61.1 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.11.2 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.1 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.41.0 (active)
bracket-matcher, v0.82.1 (active)
command-palette, v0.38.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.39.0 (active)
find-and-replace, v0.201.0 (inactive)
fuzzy-finder, v1.3.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.1 (active)
image-view, v0.58.2 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.1 (inactive)
markdown-preview, v0.158.0 (active)
metrics, v0.53.1 (active)
notifications, v0.65.0 (active)
open-on-github, v1.2.0 (inactive)
package-generator, v1.0.0 (inactive)
settings-view, v0.242.2 (active)
snippets, v1.0.2 (active)
spell-check, v0.67.1 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.0 (active)
symbols-view, v0.113.0 (inactive)
tabs, v0.100.2 (active)
timecop, v0.33.2 (active)
tree-view, v0.208.2 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.34.0 (active)
whitespace, v0.33.0 (active)
wrap-guide, v0.38.1 (active)
language-c, v0.52.1 (active)
language-clojure, v0.21.0 (active)
language-coffee-script, v0.47.2 (active)
language-csharp, v0.12.1 (active)
language-css, v0.37.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.42.1 (active)
language-html, v0.45.1 (active)
language-hyperlink, v0.16.0 (active)
language-java, v0.23.0 (active)
language-javascript, v0.119.0 (active)
language-json, v0.18.2 (active)
language-less, v0.29.5 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.35.0 (active)
language-php, v0.37.2 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.0 (active)
language-ruby, v0.69.0 (active)
language-ruby-on-rails, v0.25.0 (active)
language-sass, v0.56.0 (active)
language-shellscript, v0.22.4 (active)
language-source, v0.9.0 (active)
language-sql, v0.23.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.28.0 (active)
language-toml, v0.18.0 (active)
language-xml, v0.34.9 (active)
language-yaml, v0.26.0 (active)

# Dev
No dev packages
```

**Comment by ddenisenko** [comment]
Cant reproduce.

Here is RAML, I tried it with:
```
#%RAML 1.0
title: test
types:
  Pet:
    properties:
      name: string
      kind: string
      price: number
    example:
      name: "Snoopy"
      kind: "Mammal"
      price: 100

  Mammal:
    type: Pet
    properties:
      color:
        enum: [White, Black, Colored]

  Bird:
    type: Pet
    properties:
      wingLength: number

```

Steps:
1. Left-click on `Mammal:`. 
2. Right-click, Context menu->Refactoring->Pull Up.
3. Select `color`, click on a button `>`.
4. Click on `extract` button.

Is that correct steps?

**Comment by drernie** [comment]
Thanks for getting back so quickly!

When I tried it with your file, it worked properly.  However, during the tutorial there's additional content below the types:
```
#%RAML 1.0
title: Pet shop
version: 1
baseUri: /shop
types:
  Pet:
    properties:
      name: string
      kind: string
      price: number
    example:
      name: "Snoopy"
      kind: "Mammal"
      price: 100

  Mammal:
    type: Pet
    properties:
      color:
        enum: [White, Black, Colored]

  Bird:
    type: Pet
    properties:
      wingLength: number
       
resourceTypes:

  Collection:
    get:
      responses:
        200:
          body:
            application/json:
              type: <<item>>[]
    post:
      body:
        application/json:
          type: <<item>>

  Member:
    put:
      body:
        application/json:
          type: <<item>>
    delete:
      responses:
        204:

traits:
  FilterableByPrice:
    queryParameters:
      priceLessThen?:
        type: number
      priceMoreThen?:
        type: number

/pets:
  type: { Collection: {item: Pet} }
  get:
    is: FilterableByPrice
    queryParameters:
      petKind:
        enum: [bird, mammal]
  /{id}:
    type: { Member: {item : Pet} }
```
When I try a Pull Up on that file I still get that error.
(Note that I am running the new version of macOS, in case that matters).

**Comment by ddenisenko** [comment]
It is reproducible with the latest variant of RAML content, thank you.

**Comment by dreamflyer** [comment]
will be published in v0.8.40.
